### PR TITLE
fix(block-diff): 修复区块发布是查看差异保存时schema出错的bug

### PR DESCRIPTION
**Pull request by betterdancing**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?
保存时由于newCode为空字符串，导致schema出错

Issue Number: N/A

### What is the new behavior?
当newCode为空时，不更新任何内容

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by wenmine — DISMISSED** [review]
LGTM

File: packages/common/component/BlockDeployDialog.vue [hunk 1]
```diff
@@ -212,10 +212,20 @@ export default {
     }
 
     const save = () => {
-      const pageSchema = JSON.parse(state.newCode)
-      setSchema(pageSchema)
-
-      close()
+      if (state.newCode) {
```
Comment (chilingling) [inline packages/common/component/BlockDeployDialog.vue]: 使用 earlyReturn，
```javascript
if (!state.newCode) {
  close();
  return;
}
try {
 ...
}
```

File: packages/common/component/BlockDeployDialog.vue [hunk 2]
```diff
@@ -212,10 +212,20 @@ export default {
...
+        try {
+          const pageSchema = JSON.parse(state.newCode)
+          setSchema(pageSchema)
```
Comment (chilingling) [inline packages/common/component/BlockDeployDialog.vue]: 这里 componentName 不允许修改，componentName 被用来区分页面和区块的逻辑，允许修改会有很多问题。

```
setSchema({
  ...pageSchema,
 componentName:  originComponentName
})
```

Comment (betterdancing) [reply packages/common/component/BlockDeployDialog.vue]: > 这里 componentName 不允许修改，componentName 被用来区分页面和区块的逻辑，允许修改会有很多问题。
> 
> ```
> setSchema({
>   ...pageSchema,
>  componentName:  originComponentName
> })
> ```

已修改

**Review by chilingling — APPROVED** [review]

**Pull request merged**